### DISCUSSION TITLE
sqlcipher_flutter_libs: file is not a database, file is not a database (code 26)

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#2687] [General]
Bit of an odd one this.

Released mutiple versions of our app, always internal -> Open Beta -> Production

Released an update yesterday (DB) code hasn't changed at all and I get the following and it won't boot the app. 

SqliteException(26): while executing, file is not a database, file is not a database (code 26) Causing statement: pragma journal_mode=WAL;, parameters:

This is updating an existing  installed app. I have tried downloading the previous apk and then updating it but I can't reproduce this, makes me hesitant to push to release.

Not sure if I'm doing anything wrong but this has been working fine.

```yaml
  sqlite3: ^2.1.0
  sqlcipher_flutter_libs: ^0.6.0
  drift: ^2.12.1
```

```dart
DatabaseConnection.delayed(() async {
        final isolate = await _createDriftIsolate();
        return await isolate.connect(
          singleClientMode: true,
        );
      }
```

```dart
Future<DriftIsolate> _createDriftIsolate() async {
  final token = RootIsolateToken.instance;
  return await DriftIsolate.spawn(() {
    BackgroundIsolateBinaryMessenger.ensureInitialized(token!);

    if (Platform.isAndroid) {
      open.overrideFor(
        OperatingSystem.android,
        () => DynamicLibrary.open('libsqlcipher.so'),
      );
    }

    return LazyDatabase(() async {
      final dbFolder = await getApplicationDocumentsDirectory();
      final path = p.join(dbFolder.path, 'bc.db');

      var entityKey = await SecuredStorage.instance
          .getStringValue(SharedAppConfig.entityKey);

      return NativeDatabase(
        File(path),
        logStatements: false,
        setup: (db) {
          final result = db.select('pragma cipher_version');
          if (result.isEmpty) {
            throw UnsupportedError(
              'This database needs to run with SQLCipher, but that library is '
              'not available!',
            );
          }

          db
            ..execute("pragma key = '$entityKey'")
            ..execute('pragma journal_mode=WAL;')
            ..execute('pragma busy_timeout = 100;');
        },
      );
    });
  });
}
```

**Comment by Foxsterc** [comment]
This is now happening on both Android and iOS, randomly. If it helps this is the trace.

`0
communication.dart - Line 113
DriftCommunication.request + 113
1
client_impl.dart - Line 158
_RemoteQueryExecutor.ensureOpen + 158
2
lazy_database.dart - Line 61
LazyDatabase.ensureOpen.<fn> + 61
3
query.dart - Line 240
Selectable.getSingleOrNull + 240
4
user_service.dart - Line 34
UserService.first + 34
5
authentication_bloc.dart - Line 55
AuthenticationBloc._onAuthenticate + 55
6
bloc.dart - Line 229
Bloc.on.<fn>.handleEvent + 229`

**Comment by simolus3** [comment]
Are you sure that nothing could have changed the return value of `SecuredStorage.instance.getStringValue(SharedAppConfig.entityKey)`? An invalid key would seem to be the most likely cause of that failure.

**Reply by Foxsterc** [reply]
Highly possible of course! I'll revisit my code to see if I can find anything. Thanks for the pointer.
